### ISSUE TITLE
[CLI/API] When the CLI command `empty-upload --force` is used on a repo with carry forward flags, the notifications fail

### ISSUE THREAD

**Issue by drazisil-codecov** [codecov/engineering-team#2365]
When the CLI command `empty-upload --force` is used on a repo with carry forward flags, Notifications fail due to the notifications task being called by API before the report has a chance to be populated with the report totals

![Image](https://github.com/user-attachments/assets/e06065a4-28e8-4e4e-aec5-f12ce77f0c79)

**Comment by eliatcodecov** [comment]
Note that `--force` isn't documented and is only discoverable in code. At a minimum we should document this and state that it doesn't work with carryforward flags. 

More generally, we should align on why this feature exists, if it should exist, and if so how to document it. It appears this was just  implemented as a convenience method to send notifications without necessarily having to call send notification manually. Do we even to have this flag?

**Comment by erickpeirson** [comment]
I believe that we are impacted by this issue. We use `empty-upload` to ensure that we trigger a CodeCov report regardless of whether CI has generated coverage data, so that we can globally require the `codecov/patch` status to merge pull requests. 

We use carry-forward flags extensively.

Note that without the `--force` flag, calls to `empty-upload` fail with:

```
info - 2024-09-10 18:24:39,260 -- Process Empty Upload complete
debug - 2024-09-10 18:24:39,260 -- Empty Upload result --- {"result": "RequestResult(error=RequestError(code='HTTP Error 404', params={}, description='{\"detail\":\"Unable to get pull request for commit: 19cc963c9bda826fe1a8e4d2e00bf8b05cede023\"}'), warnings=[], status_code=404, text='{\"detail\":\"Unable to get pull request for commit: 19cc963c9bda826fe1a8e4d2e00bf8b05cede023\"}')"}
error - 2024-09-10 18:24:39,260 -- Empty Upload failed: {"detail":"Unable to get pull request for commit: 19cc963c9bda826fe1a8e4d2e00bf8b05cede023"}
```

What is the correct way to ensure that CodeCov always generates a report on PRs in a project that relies on carry-forward flags?

**Comment by drazisil-codecov** [comment]
Hi @erickpeirson,

Can you please try calling `create-commit` before `empty-upload`?

You _might_ need to call `create-report` in between those two, but I _think_ `empty-upload` handles the same thing.

You should then get your notifications after we generate the "empty" report with all the carried forward coverage data.

**Comment by erickpeirson** [comment]
> Can you please try calling `create-commit` before `empty-upload`?

Yes, we are already doing this today

> You _might_ need to call `create-report` in between those two, but I _think_ `empty-upload` handles the same thing.

This appears to have done the trick!

**Comment by drazisil-codecov** [comment]
(Forgot this was was a bug ticket and not a customer one 🤦‍♀

**Comment by erickpeirson** [comment]
@drazisil-codecov bad news.... this is now more broken than before. This is having a pretty big impact on CI for us.

```

debug - 2024-09-12 20:15:37,714 -- Using ci service from provider name: BuildKite
--
  | debug - 2024-09-12 20:15:37,715 -- versioning system found: <class 'codecov_cli.helpers.versioning_systems.GitVersioningSystem'>
  | debug - 2024-09-12 20:15:37,717 -- versioning system found: <class 'codecov_cli.helpers.versioning_systems.GitVersioningSystem'>
  | debug - 2024-09-12 20:15:37,718 -- Loading config from /workdir/codecov.yml
  | debug - 2024-09-12 20:15:37,735 -- Starting create commit process --- {"commit_sha": "aa202051e59e09d04ca3f2e8b08069f6ba9a6148", "parent_sha": null, "pr": null, "branch": "main", "slug": "datagravity-ai/dquality", "token": "6******************", "service": "github", "enterprise_url": null}
  | info - 2024-09-12 20:15:37,900 -- Process Commit creating complete
  | debug - 2024-09-12 20:15:37,900 -- Commit creating result --- {"result": "RequestResult(error=None, warnings=[], status_code=201, text='{\"message\":\"...\",\"timestamp\":\"2024-09-12T15:53:32Z\",\"ci_passed\":false,\"state\":\"complete\",\"repository\":{\"name\":\"dquality\",\"is_private\":true,\"active\":true,\"language\":\"python\",\"yaml\":{\"ignore\":[\"^ai_lib/sql_fixer.*\",\"^ai_lib/sql_generator.*\",\"^www/manage.py.*\",\"(?s:www/site_apps/main/migrations/[^\\\\\\\\/]*_auto_[^\\\\\\\\/]*\\\\\\\\.py)\\\\\\\\Z\",\"^www/tests.*\"],\"comment\":{\"layout\":\"reach, diff, components, files\",\"show_carryforward_flags\":false},\"flag_management\":{\"default_rules\":{\"statuses\":[{\"base\":\"auto\",\"type\":\"project\",\"target\":\"auto\",\"threshold\":1.0,\"if_ci_failed\":\"error\"},{\"type\":\"patch\",\"target\":100.0,\"if_ci_failed\":\"error\"}],\"carryforward\":true},\"individual_flags\":[{\"name\":\"dquality\",\"paths\":[\".*\"],\"carryforward\":true},{\"name\":\"backtests\",\"paths\":[\".*\"],\"carryforward\":true},{\"name\":\"shared\",\"paths\":[\"shared/.*\"],\"carryforward\":true},{\"name\":\"web-backend\",\"paths\":[\"www/.*\"],\"carryforward\":true}]},\"component_management\":{\"individual_components\":[{\"name\":\"Data warehouse connectors\",\"paths\":[\"^dquality/db.*\"],\"component_id\":\"dbconnectors\"},{\"name\":\"Data quality core\",\"paths\":[\"^dquality.*\"],\"component_id\":\"dquality\"},{\"name\":\"Backend web app\",\"paths\":[\"^www.*\"],\"component_id\":\"backend\"},{\"name\":\"Shared library\",\"paths\":[\"^shared.*\"],\"component_id\":\"shared\"},{\"name\":\"CLI apps\",\"paths\":[\"^apps.*\"],\"component_id\":\"apps\"}]}}},\"author\":{\"avatar_url\":\"https://avatars0.githubusercontent.com/u/6236330?v=3&s=55\",\"service\":\"github\",\"username\":\"kfoxb\",\"name\":\"Kyle Bradford\",\"ownerid\":3670035},\"commitid\":\"aa202051e59e09d04ca3f2e8b08069f6ba9a6148\",\"parent_commit_id\":\"309fd0073938a8a90743e84fb53e14b1161013c6\",\"pullid\":null,\"branch\":\"main\"}')"}
  | debug - 2024-09-12 20:15:38,179 -- Using ci service from provider name: BuildKite
  | debug - 2024-09-12 20:15:38,181 -- versioning system found: <class 'codecov_cli.helpers.versioning_systems.GitVersioningSystem'>
  | debug - 2024-09-12 20:15:38,182 -- versioning system found: <class 'codecov_cli.helpers.versioning_systems.GitVersioningSystem'>
  | debug - 2024-09-12 20:15:38,183 -- Loading config from /workdir/codecov.yml
  | debug - 2024-09-12 20:15:38,193 -- Starting create report process --- {"commit_sha": "aa202051e59e09d04ca3f2e8b08069f6ba9a6148", "code": "default", "slug": "datagravity-ai/dquality", "service": "github", "enterprise_url": null, "token": "6******************"}
  | info - 2024-09-12 20:15:38,320 -- Process Report creating complete
  | debug - 2024-09-12 20:15:38,320 -- Report creating result --- {"result": "RequestResult(error=None, warnings=[], status_code=201, text='{\"code\":null}')"}
  | info - 2024-09-12 20:15:38,320 -- Finished creating report successfully --- {"response": "{\"code\":null}"}
  | debug - 2024-09-12 20:15:38,598 -- Using ci service from provider name: BuildKite
  | debug - 2024-09-12 20:15:38,600 -- versioning system found: <class 'codecov_cli.helpers.versioning_systems.GitVersioningSystem'>
  | debug - 2024-09-12 20:15:38,601 -- versioning system found: <class 'codecov_cli.helpers.versioning_systems.GitVersioningSystem'>
  | debug - 2024-09-12 20:15:38,602 -- Loading config from /workdir/codecov.yml
  | debug - 2024-09-12 20:15:38,612 -- Starting empty upload process --- {"commit_sha": "aa202051e59e09d04ca3f2e8b08069f6ba9a6148", "slug": "datagravity-ai/dquality", "token": "6******************", "service": "github", "enterprise_url": null, "fail_on_error": false}
  | warning - 2024-09-12 20:15:38,765 -- Response status code was 500. --- {"retry": 0}
  | warning - 2024-09-12 20:15:38,765 -- Request failed. Retrying --- {"retry": 0}
  | warning - 2024-09-12 20:15:39,414 -- Response status code was 500. --- {"retry": 1}
  | warning - 2024-09-12 20:15:39,414 -- Request failed. Retrying --- {"retry": 1}
  | warning - 2024-09-12 20:15:40,588 -- Response status code was 500. --- {"retry": 2}
  | warning - 2024-09-12 20:15:40,588 -- Request failed. Retrying --- {"retry": 2}
  | Traceback (most recent call last):
  | File "/usr/local/bin/codecovcli", line 8, in <module>
  | sys.exit(run())
  | File "/usr/local/lib/python3.10/site-packages/codecov_cli/main.py", line 81, in run
  | cli(obj={})
  | File "/usr/local/lib/python3.10/site-packages/click/core.py", line 1157, in __call__
  | return self.main(*args, **kwargs)
  | File "/usr/local/lib/python3.10/site-packages/click/core.py", line 1078, in main
  | rv = self.invoke(ctx)
  | File "/usr/local/lib/python3.10/site-packages/click/core.py", line 1688, in invoke
  | return _process_result(sub_ctx.command.invoke(sub_ctx))
  | File "/usr/local/lib/python3.10/site-packages/click/core.py", line 1434, in invoke
  | return ctx.invoke(self.callback, **ctx.params)
  | File "/usr/local/lib/python3.10/site-packages/click/core.py", line 783, in invoke
  | return __callback(*args, **kwargs)
  | File "/usr/local/lib/python3.10/site-packages/click/decorators.py", line 33, in new_func
  | return f(get_current_context(), *args, **kwargs)
  | File "/usr/local/lib/python3.10/site-packages/codecov_cli/commands/empty_upload.py", line 41, in empty_upload
  | return empty_upload_logic(
  | File "/usr/local/lib/python3.10/site-packages/codecov_cli/services/empty_upload/__init__.py", line 22, in empty_upload_logic
  | sending_result = send_post_request(
  | File "/usr/local/lib/python3.10/site-packages/codecov_cli/helpers/request.py", line 82, in wrapper
  | raise Exception("Request failed after too many retries")
  | Exception: Request failed after too many retries


```

**Comment by drazisil-codecov** [comment]
Hi @erickpeirson ,

We were trying to fix it last night, sorry that it broke. This was with `--force`, or no?

**Comment by erickpeirson** [comment]
> Hi @erickpeirson ,
> 
> We were trying to fix it last night, sorry that it broke. This was with `--force`, or no?

Without `--force`

**Comment by drazisil-codecov** [comment]
Is it still broken? _runs a test to see_

**Comment by drazisil-codecov** [comment]
Ok, very glad I have an engineer in my timezone. Investigating

**Comment by drazisil-codecov** [comment]
Fix PR for the 500 (not `--force`) is running CI.

**Comment by drazisil-codecov** [comment]
@erickpeirson the 500 is fixed.

**Comment by drazisil-codecov** [comment]
Hooky, stop closing this.

**Comment by giovanni-guidini** [comment]
Is [this](https://github.com/codecov/engineering-team/issues/2365#issue-2478428094) quote still accurate? (from the 1st comment)
That is, is that the bug?

> When the CLI command empty-upload --force is used on a repo with carry forward flags, Notifications fail due to the notifications task being called by API before the report has a chance to be populated with the report totals

**Comment by drazisil-codecov** [comment]
@giovanni-guidini I believe this is still the behavior on public repos, where creds are not needed.

I haven't see it recently though , so I guess we can close.

**Comment by drazisil-codecov** [comment]
This is still an issue after the creds issue was fixed.

![Image](https://github.com/user-attachments/assets/59436b1d-e31f-4608-bce0-58224a71b95b)

**Comment by eliatcodecov** [comment]
To sum: We appear to crash the notifier if we're using `empty-upload` in conjunction with carryforward flags. This leads to checks stuck eternally in the pending state. 

We should investigate the combination of these two features and see if we can resolve this issue.

**Comment by jhecking** [comment]
@eliatcodecov @drazisil-codecov Any possible work-around we could do on our end in the meantime?

**Comment by jhecking** [comment]
@drazisil-codecov, in https://github.com/codecov/feedback/issues/108#issuecomment-2419948031 you mentioned "I think we are almost there" in relation to this ticket. Any further updates you can share? Any possible work-around we can do on our end in the meantime?

**Comment by drazisil-codecov** [comment]
@jhecking we are trying combos of https://github.com/codecov/engineering-team/issues/2365#issuecomment-2341902606 and it's mostly workig. I think you said that didn't for you, but if you would like to send me a new SHA, I can take another look. Please do NOT pass `--force`.

We are actively working on this, I understand the frustration and really appreciate your patience .

**Comment by jhecking** [comment]
@drazisil-codecov we currently _don't_ use `--force`. Here is the sequence of commands we execute:

```
  curl -Os https://cli.codecov.io/v0.7.5/linux/codecov
  chmod +x codecov
  ./codecov create-commit
  ./codecov create-report
  ./codecov empty-upload
```

**Comment by drazisil-codecov** [comment]
@jhecking I'm trying look up that SH, but GCP logs are runninh very slow. Still looking.

**Comment by jhecking** [comment]
@drazisil-codecov Not sure what to do with that comment. But let me know if you need more SHAs. Had to help the team merge half-a-dozen PRs today that got stuck because of this issue... :-(

**Comment by drazisil-codecov** [comment]
Can you please share the newest one you have? And let me know the date?, @jhecking

**Comment by jhecking** [comment]
https://github.com/Securezapp/borneo/pull/3648/commits/82d489acc702eb9990fce508e57c9094c03ef17f - 2024-10-22T08:52:14Z

**Comment by drazisil-codecov** [comment]
> https://github.com/Securezapp/borneo

**Comment by drazisil-codecov** [comment]
Wrong button

**Comment by drazisil-codecov** [comment]
> https://github.com/Securezapp/borneo/pull/3644/commits/2cc4bb6799ef5f0ae782c2582ec325ca954dbead - 2024-10-22T06:49:10.000Z

Can you share the uploader output from this SHA? I'm not sure it uploaded as expected.

Also, are you using the uploader, or the CLI? I may have confused issues.

**Comment by jhecking** [comment]
https://github.com/Securezapp/borneo/pull/3635/commits/973c5757d3bd40b3a96bb33c9aac228914374c86

https://app.codecov.io/gh/Securezapp/borneo/commit/973c5757d3bd40b3a96bb33c9aac228914374c86



```
2024-10-21T22:42:30.5632209Z ##[group]Run curl -Os https://cli.codecov.io/v0.7.5/linux/codecov
2024-10-21T22:42:30.5632798Z [36;1mcurl -Os https://cli.codecov.io/v0.7.5/linux/codecov[0m
2024-10-21T22:42:30.5633469Z [36;1mchmod +x codecov[0m
2024-10-21T22:42:30.5633808Z [36;1m./codecov create-commit[0m
2024-10-21T22:42:30.5634181Z [36;1m./codecov create-report[0m
2024-10-21T22:42:30.5634536Z [36;1m./codecov empty-upload[0m
2024-10-21T22:42:30.5655967Z shell: /usr/bin/bash -e {0}
2024-10-21T22:42:30.5656332Z env:
2024-10-21T22:42:30.5664119Z   NX_BASE: 6c108150888c46d78bb180701e40c0f9c1ff27d7
2024-10-21T22:42:30.5664587Z   NX_HEAD: 7c76f426c157e2df4b87a6494476eccc8950ab0a
2024-10-21T22:42:30.5665025Z   NPM_CONFIG_USERCONFIG: /home/runner/work/_temp/.npmrc
2024-10-21T22:42:30.5665470Z   NODE_AUTH_TOKEN: XXXXX-XXXXX-XXXXX-XXXXX
2024-10-21T22:42:30.5675554Z   GITHUB_DEPENDENCY_GRAPH_ENABLED: false
2024-10-21T22:42:30.5675995Z   CODECOV_TOKEN: ***
2024-10-21T22:42:30.5676311Z ##[endgroup]
2024-10-21T22:42:31.7066130Z info - 2024-10-21 22:42:31,706 -- ci service found: github-actions
2024-10-21T22:42:37.9220696Z info - 2024-10-21 22:42:37,921 -- Process Commit creating complete
2024-10-21T22:42:38.5230486Z info - 2024-10-21 22:42:38,522 -- ci service found: github-actions
2024-10-21T22:42:44.6994059Z info - 2024-10-21 22:42:44,698 -- Process Report creating complete
2024-10-21T22:42:45.3045394Z info - 2024-10-21 22:42:44,699 -- Finished creating report successfully --- {"response": "{\"status\":\"queued\"}\n"}
2024-10-21T22:42:45.3046279Z info - 2024-10-21 22:42:45,304 -- ci service found: github-actions
2024-10-21T22:42:51.4941132Z info - 2024-10-21 22:42:51,493 -- Process Empty Upload complete
```

![Image](https://github.com/user-attachments/assets/3c120153-246f-4010-b380-24d6328d9e8c)

![Image](https://github.com/user-attachments/assets/c4621d5b-0ae3-4b02-908b-93c4cb6e4e6e)

**Comment by drazisil-codecov** [comment]
@jhecking 

I have too many similar issues around empty upload, my apologies for the confusion. This is a bug (that's I'm going to double check I created an issue for. Can you please try adding `--force` to your empty upload commands and see if that changes them to success?

I initially thought you weren't getting them, but we successfully sent the checks...or at least GitHub told us we did

![Image](https://github.com/user-attachments/assets/a4cdf0f6-8f79-4d10-9f35-7aa76f75b93e)

**Comment by jhecking** [comment]
@drazisil-codecov 

No difference when using `--force` either:

https://github.com/Securezapp/borneo/pull/3530/commits/


```
2024-10-23T03:17:03.5212874Z ##[group]Run curl -Os https://cli.codecov.io/v0.7.5/linux/codecov
2024-10-23T03:17:03.5213444Z [36;1mcurl -Os https://cli.codecov.io/v0.7.5/linux/codecov[0m
2024-10-23T03:17:03.5213876Z [36;1mchmod +x codecov[0m
2024-10-23T03:17:03.5214214Z [36;1m./codecov create-commit[0m
2024-10-23T03:17:03.5214575Z [36;1m./codecov create-report[0m
2024-10-23T03:17:03.5214923Z [36;1m./codecov empty-upload --force[0m
2024-10-23T03:17:03.5236208Z shell: /usr/bin/bash -e {0}
2024-10-23T03:17:03.5236535Z env:
2024-10-23T03:17:03.5244091Z   NX_BASE: 7d9e82dff2bde0ba352c5270ffa0167e4fb9c35c
2024-10-23T03:17:03.5244550Z   NX_HEAD: 907a599c62b9af3996b6f8accd102fa1860f0b60
2024-10-23T03:17:03.5244997Z   NPM_CONFIG_USERCONFIG: /home/runner/work/_temp/.npmrc
2024-10-23T03:17:03.5245430Z   NODE_AUTH_TOKEN: XXXXX-XXXXX-XXXXX-XXXXX
2024-10-23T03:17:03.5255683Z   CODECOV_TOKEN: ***
2024-10-23T03:17:03.5255984Z ##[endgroup]
2024-10-23T03:17:05.1080546Z info - 2024-10-23 03:17:05,107 -- ci service found: github-actions
2024-10-23T03:17:05.3372370Z info - 2024-10-23 03:17:05,336 -- Process Commit creating complete
2024-10-23T03:17:05.9380167Z info - 2024-10-23 03:17:05,937 -- ci service found: github-actions
2024-10-23T03:17:06.1155883Z info - 2024-10-23 03:17:06,115 -- Process Report creating complete
2024-10-23T03:17:06.7185804Z info - 2024-10-23 03:17:06,115 -- Finished creating report successfully --- {"response": "{\"status\":\"queued\"}\n"}
2024-10-23T03:17:06.7186852Z info - 2024-10-23 03:17:06,718 -- ci service found: github-actions
2024-10-23T03:17:06.8963522Z info - 2024-10-23 03:17:06,895 -- Process Empty Upload complete
```

![Image](https://github.com/user-attachments/assets/a3745fb9-9355-40cc-94e8-04c18caddb2a)

---

This failed commit status "Empty Upload" is a fairly recent change. Nothing has changed on our end, but something must have changed on your end. Because we did not used to get this. Instead we used to see this status message "waiting for status to be reported":

![Image](https://github.com/user-attachments/assets/f507a5db-3186-4667-bb3f-3151b8bfe3c0)

**Comment by jhecking** [comment]
@drazisil-codecov  @eliatcodecov @trent-codecov 

> I have too many similar issues around empty upload

That doesn't inspire me with confidence that this feature will work as expected any time soon.

I've been thinking, what if we create a dummy project with static code coverage results in our monorepo instead. And if none of the other projects create any coverage results, then we instead submit the (static) results for this dummy project? Kinda like an empty upload but for a "real" project instead? Would that be a feasible work-around until you can get your act together and ship a working version of this feature?

We're reaching the end of our patience on this issue, which has been ongoing for more than a month at this point...

**Comment by jhecking** [comment]
Here is an example of a PR/commit where we still see the "codecov/patch - Waiting for status to be reported" status even now. This is using the same Github workflow with the exact same sequence of codecov CLI commands (without `--force`). I have no clue why we sometimes see a failed "Empty upload" status and other times "waiting for status to be reported".

https://github.com/Securezapp/borneo/pull/3668/commits/899035b7b6e63a5bc7a5a47949f172b6ce7a54ad

```
2024-10-23T17:24:59.1358186Z ##[group]Run curl -Os https://cli.codecov.io/v0.7.5/linux/codecov
2024-10-23T17:24:59.1358736Z [36;1mcurl -Os https://cli.codecov.io/v0.7.5/linux/codecov[0m
2024-10-23T17:24:59.1359164Z [36;1mchmod +x codecov[0m
2024-10-23T17:24:59.1359720Z [36;1m./codecov create-commit[0m
2024-10-23T17:24:59.1360105Z [36;1m./codecov create-report[0m
2024-10-23T17:24:59.1360594Z [36;1m./codecov empty-upload[0m
2024-10-23T17:24:59.1372876Z shell: /usr/bin/bash -e {0}
2024-10-23T17:24:59.1373179Z env:
2024-10-23T17:24:59.1379473Z   NX_BASE: c720611bcc0a1c17a7c068e769537a2669fd5a5f
2024-10-23T17:24:59.1379930Z   NX_HEAD: 14c44feb386d6ba5f1684d82a560e0624502951c
2024-10-23T17:24:59.1380470Z   NPM_CONFIG_USERCONFIG: /home/runner/work/_temp/.npmrc
2024-10-23T17:24:59.1380896Z   NODE_AUTH_TOKEN: XXXXX-XXXXX-XXXXX-XXXXX
2024-10-23T17:24:59.1390803Z   CODECOV_TOKEN: ***
2024-10-23T17:24:59.1391100Z ##[endgroup]
2024-10-23T17:25:00.2440558Z info - 2024-10-23 17:25:00,243 -- ci service found: github-actions
2024-10-23T17:25:00.4889429Z info - 2024-10-23 17:25:00,488 -- Process Commit creating complete
2024-10-23T17:25:01.1095973Z info - 2024-10-23 17:25:01,109 -- ci service found: github-actions
2024-10-23T17:25:01.2799043Z info - 2024-10-23 17:25:01,279 -- Process Report creating complete
2024-10-23T17:25:01.2801015Z info - 2024-10-23 17:25:01,279 -- Finished creating report successfully --- {"response": "{\"status\":\"queued\"}\n"}
2024-10-23T17:25:01.9017827Z info - 2024-10-23 17:25:01,901 -- ci service found: github-actions
2024-10-23T17:25:02.1433571Z info - 2024-10-23 17:25:02,142 -- Process Empty Upload complete
```

![Image](https://github.com/user-attachments/assets/bfdc8a70-6cb2-4693-92de-0632596645eb)

**Comment by Swatinem** [comment]
Hi @jhecking 

The `--force` flag was undocumented so far, which I am rectifying here: https://github.com/codecov/codecov-cli/pull/543

TLDR:
You will always get a "passing" status check when using `--force`. Otherwise, codecov will actually look at the PR diff and determine the changed files, marking the check as "failing" if it detects any files being changed in a PR that are *not* ignored via the `codecov.yml` file.

So for your usecase, using `--force` is the right thing to do, as you want to have a passing check, no matter what.

---

As for the checks sometimes not triggering at all, I am still investigating this. There might be some cases in which notifications (aka updating the PR status) are skipped / not triggered at all.
The intention is that they should be triggered when you use `empty-upload` no matter what, and I will try to reproduce and fix the underlying issue.

**Comment by Swatinem** [comment]
As another note, I would expect there to be more log output after the `Process Empty Upload complete` line, but your copy-pasted output does not include that.
If there is indeed no further output, that means that the API response does not result in a `200` status code, and we would have to figure out why that is.

https://github.com/codecov/codecov-cli/blob/6269e67107ed0bc2353d1bf89dfe1c19ffd2c09e/codecov_cli/services/empty_upload/__init__.py#L38-L41

**Comment by Swatinem** [comment]
There were indeed a bunch of errors related to missing coverage reports, which I believe was also mentioned above as races between `create-report` and `empty-upload`.

The fix is here: https://github.com/codecov/worker/pull/816
Which fixes this internal error: https://codecov.sentry.io/issues/5801299347/events/475fc10aeedd4005a6162a55b90d638a/?project=4165036&statsPeriod=7d (@drazisil-codecov has access)
… which was raised internally for your commit https://github.com/Securezapp/borneo/pull/3668/commits/899035b7b6e63a5bc7a5a47949f172b6ce7a54ad

Once this fix lands in production, this should hopefully fix (or at least reduce the likelihood of) cases where the status check is completely stuck and does not report any status.
Together with my comment above clarifying that you have to add `--force`, I think this should fix your usecase.

**Comment by jhecking** [comment]
@Swatinem 

> You will always get a "passing" status check when using --force.

That’s not the case based on what I’m seeing. Please refer to https://github.com/codecov/engineering-team/issues/2365#issuecomment-2430777732 for an example of a commit where we did use `--force` and still got a failed status check. (Just focus on the part above the horizontal line and ignore the 2nd part below.)

**Comment by jhecking** [comment]
@Swatinem 

> The fix is here: https://github.com/codecov/worker/pull/816

This PR was merged 18 hours ago. When can we expect this fix to land in production?

**Comment by Swatinem** [comment]
This should have deployed by now. Please make sure you do use the `--force` flag.

Interestingly enough, I got two traces that trigger notifications for the commit you posted above that I was checking (https://codecov.sentry.io/performance/summary/?project=4165036&query=commit_sha%3A899035b7b6e63a5bc7a5a47949f172b6ce7a54ad&referrer=performance-transaction-summary&showTransactions=recent&statsPeriod=7d&transaction=app.tasks.notify.Notify&unselectedSeries=p100%28%29&unselectedSeries=avg%28%29)

I would have expected only one such call. The second one came in a couple of hours after the first. Its quite possible these were interfering with one another.

**Comment by jhecking** [comment]
@Swatinem 

> This should have deployed by now. Please make sure you do use the --force flag.

Still not working:

https://github.com/Securezapp/borneo/pull/3711/commits/095405c4b64b7e15a6ee29d51d5d020ddad0efe6

```
2024-10-28T11:26:47.4633612Z ##[group]Run curl -Os https://cli.codecov.io/v0.7.5/linux/codecov
2024-10-28T11:26:47.4634182Z [36;1mcurl -Os https://cli.codecov.io/v0.7.5/linux/codecov[0m
2024-10-28T11:26:47.4634841Z [36;1mchmod +x codecov[0m
2024-10-28T11:26:47.4635184Z [36;1m./codecov create-commit[0m
2024-10-28T11:26:47.4635530Z [36;1m./codecov create-report[0m
2024-10-28T11:26:47.4635875Z [36;1m./codecov empty-upload --force[0m
2024-10-28T11:26:47.4657126Z shell: /usr/bin/bash -e {0}
2024-10-28T11:26:47.4657458Z env:
2024-10-28T11:26:47.4665135Z   NX_BASE: adffc24a9c2be9b328781f7b1fd5ab4cfcbe76a8
2024-10-28T11:26:47.4665612Z   NX_HEAD: d2fdca2b3460d2f7a006cb3aa12b0a5455b9526a
2024-10-28T11:26:47.4676848Z   CODECOV_TOKEN: ***
2024-10-28T11:26:47.4677151Z ##[endgroup]
2024-10-28T11:26:48.4636256Z info - 2024-10-28 11:26:48,463 -- ci service found: github-actions
2024-10-28T11:26:48.7212258Z info - 2024-10-28 11:26:48,720 -- Process Commit creating complete
2024-10-28T11:26:49.3240608Z info - 2024-10-28 11:26:49,323 -- ci service found: github-actions
2024-10-28T11:26:49.5936489Z info - 2024-10-28 11:26:49,593 -- Process Report creating complete
2024-10-28T11:26:49.5937440Z info - 2024-10-28 11:26:49,593 -- Finished creating report successfully --- {"response": "{\"status\":\"queued\"}\n"}
2024-10-28T11:26:50.2057432Z info - 2024-10-28 11:26:50,205 -- ci service found: github-actions
2024-10-28T11:26:50.4063684Z info - 2024-10-28 11:26:50,405 -- Process Empty Upload complete
```

![Image](https://github.com/user-attachments/assets/a67b722e-1189-406c-b6a7-6946ac00eac3)

**Comment by Swatinem** [comment]
Thanks for the new commit link.

The notifications that are being triggered are matching this internal trace @drazisil-codecov : https://codecov.sentry.io/performance/trace/0c53ce344f2448f6be8e804c958ea55f/?node=txn-134c47f4307f45b69551b0fbd2cb6202&project=4165036&query=commit_sha%3A095405c4b64b7e15a6ee29d51d5d020ddad0efe6&referrer=performance-transaction-summary&showTransactions=recent&source=performance_transaction_summary&statsPeriod=7d&timestamp=1730114853&transaction=app.tasks.notify.Notify&unselectedSeries=p100%28%29&unselectedSeries=avg%28%29

The interesting thing there is that the notification task is indeed being explicitly called with the following arguments:
```
commitid: 095405c4b64b7e15a6ee29d51d5d020ddad0efe6,
empty_upload: fail,
```

This means that the steps that happen before triggering the notification (cli and API calls) have already decided to trigger a `fail`ure, and we can thus narrow down our problem space.

**Comment by jhecking** [comment]
@Swatinem @drazisil-codecov any update?

For a brief moment yesterday I thought I had solved the issue by updating the codecov CLI from v0.7.5 to v0.8.0, because for this _one_ commit I got a "success" status from codecov for a commit with an empty upload: 33204eade7e46277857d018f0bfed5cc93f67c92

But that one commit seems to have been an exception, because today we are again seeing "fail" status from codecov for commits with empty uploads, e.g. 83b0a611ff72e516dd6673b192aa2f5c8674e71b

Any idea what's different between these two commits?

**Comment by drazisil-codecov** [comment]
Checking logs.

`33204eade7e46277857d018f0bfed5cc93f67c92` went though perfectly, as your observed.

`83b0a611ff72e516dd6673b192aa2f5c8674e71b`, however, threw

https://codecov.sentry.io/issues/6017331725/events/113a82440d364a5eb8ec3748125d1fa0/?project=4165036 
and 
https://codecov.sentry.io/issues/4370613101/events/048f3f76ab744aab8f4af632db2677b4/?project=4165036

@Swatinem 

Both head-related errors, similar to https://github.com/codecov/worker/pull/816

**Comment by drazisil-codecov** [comment]
@jhecking Can you share yoyr CI output for both SHAs? It looks like the passing one was passed `--force` based on logs and code

https://github.com/codecov/codecov-api/blob/f48452449f88a23715104f3bc17027d7c094563b/upload/views/empty_upload.py#L104-L107

**Comment by jhecking** [comment]
Both used `--force`:

https://github.com/Securezapp/borneo/commit/33204eade7e46277857d018f0bfed5cc93f67c92:

```
2024-10-29T06:40:29.6379443Z ##[group]Run curl -Os https://cli.codecov.io/v0.8.0/linux/codecov
2024-10-29T06:40:29.6380069Z [36;1mcurl -Os https://cli.codecov.io/v0.8.0/linux/codecov[0m
2024-10-29T06:40:29.6380619Z [36;1mchmod +x codecov[0m
2024-10-29T06:40:29.6381033Z [36;1m./codecov create-commit[0m
2024-10-29T06:40:29.6381430Z [36;1m./codecov create-report[0m
2024-10-29T06:40:29.6381871Z [36;1m./codecov empty-upload --force[0m
2024-10-29T06:40:29.6403690Z shell: /usr/bin/bash -e {0}
2024-10-29T06:40:29.6404053Z env:
2024-10-29T06:40:29.6404833Z   AWS_DEFAULT_REGION: ap-southeast-1
2024-10-29T06:40:29.6405229Z   AWS_REGION: ap-southeast-1
2024-10-29T06:40:29.6405792Z   AWS_ACCESS_KEY_ID: ***
2024-10-29T06:40:29.6406264Z   AWS_SECRET_ACCESS_KEY: ***
2024-10-29T06:40:29.6411367Z   AWS_SESSION_TOKEN: ***
2024-10-29T06:40:29.6411831Z   NX_BASE: 76fb3bce4ea833eee4b6d2841d5c1d519e12a323
2024-10-29T06:40:29.6412439Z   NX_HEAD: 94bebe7930436eeff7ebd69aab1a86f5c1ace215
2024-10-29T06:40:29.6412986Z   NPM_CONFIG_USERCONFIG: /home/runner/work/_temp/.npmrc
2024-10-29T06:40:29.6413477Z   NODE_AUTH_TOKEN: XXXXX-XXXXX-XXXXX-XXXXX
2024-10-29T06:40:29.6414110Z   NX_AFFECTED: []
2024-10-29T06:40:29.6414559Z   CODECOV_TOKEN: ***
2024-10-29T06:40:29.6414892Z ##[endgroup]
2024-10-29T06:40:30.6111873Z info - 2024-10-29 06:40:30,610 -- ci service found: github-actions
2024-10-29T06:40:30.8218113Z info - 2024-10-29 06:40:30,821 -- Process Commit creating complete
2024-10-29T06:40:31.4364201Z info - 2024-10-29 06:40:31,436 -- ci service found: github-actions
2024-10-29T06:40:31.6642260Z info - 2024-10-29 06:40:31,663 -- Process Report creating complete
2024-10-29T06:40:31.6643145Z info - 2024-10-29 06:40:31,664 -- Finished creating report successfully --- {"response": "{\"status\":\"queued\"}\n"}
2024-10-29T06:40:32.2792268Z info - 2024-10-29 06:40:32,278 -- ci service found: github-actions
2024-10-29T06:40:32.4531402Z info - 2024-10-29 06:40:32,452 -- Process Empty Upload complete
2024-10-29T06:40:32.5245283Z Post job cleanup.
2024-10-29T06:40:32.6652746Z Cache hit occurred on the primary key node-cache-Linux-x64-yarn-6367e3a2356e6facf1699cb920e7b2d7fc39320c33f5b4ae00094f1921439903, not saving cache.
```

https://github.com/Securezapp/borneo/commit/83b0a611ff72e516dd6673b192aa2f5c8674e71b:

```
2024-10-30T08:55:28.8259110Z ##[group]Run curl -Os https://cli.codecov.io/v0.8.0/linux/codecov
2024-10-30T08:55:28.8259677Z [36;1mcurl -Os https://cli.codecov.io/v0.8.0/linux/codecov[0m
2024-10-30T08:55:28.8260113Z [36;1mchmod +x codecov[0m
2024-10-30T08:55:28.8260452Z [36;1m./codecov create-commit[0m
2024-10-30T08:55:28.8260798Z [36;1m./codecov create-report[0m
2024-10-30T08:55:28.8261157Z [36;1m./codecov empty-upload --force[0m
2024-10-30T08:55:28.8282654Z shell: /usr/bin/bash -e {0}
2024-10-30T08:55:28.8283003Z env:
2024-10-30T08:55:28.8283639Z   AWS_DEFAULT_REGION: ap-southeast-1
2024-10-30T08:55:28.8283991Z   AWS_REGION: ap-southeast-1
2024-10-30T08:55:28.8284391Z   AWS_ACCESS_KEY_ID: ***
2024-10-30T08:55:28.8284838Z   AWS_SECRET_ACCESS_KEY: ***
2024-10-30T08:55:28.8289841Z   AWS_SESSION_TOKEN: ***
2024-10-30T08:55:28.8290229Z   NX_BASE: 43d7977e5c97dd97127ebaf97320c880b54e45f2
2024-10-30T08:55:28.8290685Z   NX_HEAD: 61517624a2b475c30e8e0bf28f140126407bb032
2024-10-30T08:55:28.8291117Z   NPM_CONFIG_USERCONFIG: /home/runner/work/_temp/.npmrc
2024-10-30T08:55:28.8291545Z   NODE_AUTH_TOKEN: XXXXX-XXXXX-XXXXX-XXXXX
2024-10-30T08:55:28.8298268Z   JAVA_HOME: /opt/hostedtoolcache/Java_Corretto_jdk/18.0.2-9.1/x64
2024-10-30T08:55:28.8298929Z   JAVA_HOME_18_X64: /opt/hostedtoolcache/Java_Corretto_jdk/18.0.2-9.1/x64
2024-10-30T08:55:28.8301381Z   GITHUB_DEPENDENCY_GRAPH_ENABLED: false
2024-10-30T08:55:28.8301806Z   CODECOV_TOKEN: ***
2024-10-30T08:55:28.8302114Z ##[endgroup]
2024-10-30T08:55:29.9304742Z info - 2024-10-30 08:55:29,930 -- ci service found: github-actions
2024-10-30T08:55:30.2485620Z info - 2024-10-30 08:55:30,248 -- Process Commit creating complete
2024-10-30T08:55:30.8461337Z info - 2024-10-30 08:55:30,845 -- ci service found: github-actions
2024-10-30T08:55:31.0694591Z info - 2024-10-30 08:55:31,068 -- Process Report creating complete
2024-10-30T08:55:31.0695543Z info - 2024-10-30 08:55:31,069 -- Finished creating report successfully --- {"response": "{\"status\":\"queued\"}\n"}
2024-10-30T08:55:31.6716780Z info - 2024-10-30 08:55:31,671 -- ci service found: github-actions
2024-10-30T08:55:31.8653693Z info - 2024-10-30 08:55:31,864 -- Process Empty Upload complete
```

**Comment by Ryang20718** [comment]
I had an idea which was to still upload a codecov report, but use a runtime codecov.yml file setting threshold to 100 and target to 0. However, it seems like there's some sort of weird caching going on in the codecov report where a runtime generated codecov.yml doesn't work (i.e codecov report still believes we're using the wrong codecov.yml file) https://github.com/codecov/codecov-cli/issues/548

**Comment by Ryang20718** [comment]
I'm hitting the same error as @jhecking in this run as well https://github.com/Ryang20718/cflag/actions/runs/11635089740
```
Run curl -Os https://cli.codecov.io/v0.8.0/linux/codecov
  curl -Os https://cli.codecov.io/v0.8.0/linux/codecov
  chmod +x codecov
  ./codecov -v create-commit -t ***
  ./codecov -v create-report -t ***
  ./codecov -v empty-upload -t *** --force
  shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
  env:
    CODECOV_TOKEN: ***
info - [2](https://github.com/Ryang20718/cflag/actions/runs/11635089740/job/32403694454#step:3:2)024-11-01 19:21:49,562 -- ci service found: github-actions
debug - 2024-11-01 19:21:49,565 -- versioning system found: <class 'codecov_cli.helpers.versioning_systems.GitVersioningSystem'>
debug - 2024-11-01 19:21:49,568 -- versioning system found: <class 'codecov_cli.helpers.versioning_systems.GitVersioningSystem'>
debug - 2024-11-01 19:21:49,570 -- Loading config from /home/runner/work/cflag/cflag/.codecov.yml
debug - 2024-11-01 19:21:49,581 -- Starting create commit process --- {"verbose": true, "auto_load_params_from": null, "codecov_yml_path": null, "enterprise_url": null, "version": "cli-0.8.0", "command": "create-commit", "parent_sha": null, "pull_request_number": "16", "branch": "test_mini", "commit_sha": "a606c4[3](https://github.com/Ryang20718/cflag/actions/runs/11635089740/job/32403694454#step:3:3)14528dc41381f78e2[4](https://github.com/Ryang20718/cflag/actions/runs/11635089740/job/32403694454#step:3:4)76f110d5b4ef5c9", "fail_on_error": false, "git_service": "github", "slug": "Ryang20718/cflag"}
info - 2024-11-01 19:21:49,977 -- Process Commit creating complete
debug - 2024-11-01 19:21:49,978 -- Commit creating result --- {"result": "RequestResult(error=None, warnings=[], status_code=202, text='{\"status\":\"queued\"}\\n')"}
info - 2024-11-01 19:21:[5](https://github.com/Ryang20718/cflag/actions/runs/11635089740/job/32403694454#step:3:5)0,716 -- ci service found: github-actions
debug - 2024-11-01 19:21:50,719 -- versioning system found: <class 'codecov_cli.helpers.versioning_systems.GitVersioningSystem'>
debug - 2024-11-01 19:21:50,721 -- versioning system found: <class 'codecov_cli.helpers.versioning_systems.GitVersioningSystem'>
debug - 2024-11-01 19:21:50,724 -- Loading config from /home/runner/work/cflag/cflag/.codecov.yml
debug - 2024-11-01 19:21:50,734 -- Starting create report process --- {"verbose": true, "auto_load_params_from": null, "codecov_yml_path": null, "enterprise_url": null, "version": "cli-0.8.0", "command": "create-report", "code": "default", "pull_request_number": "1[6](https://github.com/Ryang20718/cflag/actions/runs/11635089740/job/32403694454#step:3:6)", "commit_sha": "a606c4314528dc41381f78e24[7](https://github.com/Ryang20718/cflag/actions/runs/11635089740/job/32403694454#step:3:7)6f110d5b4ef5c9", "fail_on_error": false, "git_service": "github", "slug": "Ryang2071[8](https://github.com/Ryang20718/cflag/actions/runs/11635089740/job/32403694454#step:3:8)/cflag"}
info - 2024-11-01 1[9](https://github.com/Ryang20718/cflag/actions/runs/11635089740/job/32403694454#step:3:9):21:50,992 -- Process Report creating complete
debug - 2024-11-01 19:21:50,993 -- Report creating result --- {"result": "RequestResult(error=None, warnings=[], status_code=202, text='{\"status\":\"queued\"}\\n')"}
info - 2024-11-01 19:21:50,993 -- Finished creating report successfully --- {"response": "{\"status\":\"queued\"}\n"}
info - 2024-11-01 19:21:51,727 -- ci service found: github-actions
debug - 2024-11-01 19:21:51,729 -- versioning system found: <class 'codecov_cli.helpers.versioning_systems.GitVersioningSystem'>
debug - 2024-11-01 19:21:51,732 -- versioning system found: <class 'codecov_cli.helpers.versioning_systems.GitVersioningSystem'>
debug - 2024-11-01 19:21:51,734 -- Loading config from /home/runner/work/cflag/cflag/.codecov.yml
debug - 2024-11-01 19:21:51,745 -- Starting empty upload process --- {"verbose": true, "auto_load_params_from": null, "codecov_yml_path": null, "enterprise_url": null, "version": "cli-0.8.0", "command": "empty-upload", "force": true, "commit_sha": "a606c4314528dc41381f78e2476f1[10](https://github.com/Ryang20718/cflag/actions/runs/11635089740/job/32403694454#step:3:11)d5b4ef5c9", "fail_on_error": false, "git_service": "github", "slug": "Ryang20718/cflag"}
info - 2024-[11](https://github.com/Ryang20718/cflag/actions/runs/11635089740/job/32403694454#step:3:12)-01 19:21:52,182 -- Process Empty Upload complete
debug - 2024-11-01 19:21:52,182 -- Empty Upload result --- {"result": "RequestResult(error=None, warnings=[], status_code=202, text='{\"status\":\"queued\"}\\n')"}
```

**Comment by drazisil-codecov** [comment]
@Ryang20718 The logs you pasted do not show an error. https://github.com/Ryang20718/cflag/pull/16#issuecomment-2448686291 looks correct. What issue are you seeing?

**Comment by Ryang20718** [comment]
@drazisil-codecov 

![Image](https://github.com/user-attachments/assets/bf4bf5f4-d675-4388-998d-a3395a3c8f46)

I'm expecting to see these status checks as "passed" rather than empty coverage based on docs. Am I misunderstanding?

**Comment by drazisil-codecov** [comment]
@Ryang20718 You are correct, those should be pass if this was working correctly. @Swatinem please see above for a public example where `--force` is not returning successful checks.

**Comment by Ryang20718** [comment]
@drazisil-codecov @Swatinem let me know if there's any other info I can provide or if there's any way I can help push this over the finish line!

Dayjob uses an internal job that polls the codecov api to do similar functionality of those status checks which is reinventing the wheel, but we do this purely because of this lack of bypass

**Comment by drazisil-codecov** [comment]
Hi @Ryang20718 

We think we identified the problem and are waiting for the engineer to commit a PR. Shouldn't be too much longer. Sorry this was a pain to locate.

**Comment by drazisil-codecov** [comment]
@Ryang20718 https://github.com/codecov/shelter/pull/215 should have fixed it. Can you please test? 🤞

**Comment by jhecking** [comment]
@drazisil-codecov  is Ryang20718's issue the same issue that is affecting us? Or are those two separate issues?

**Comment by drazisil-codecov** [comment]
@jhecking , sorry. Yes, this should be fixed for you now as well, I believe it was the same issue. Please remember to use `--force` for a passing check.

**Comment by Ryang20718** [comment]
@drazisil-codecov 

This is still failing (status checks aren't bypassed and just never reported). the changes that fix the issue are server side right?

![Image](https://github.com/user-attachments/assets/fa2b1b27-2716-4f97-8dbe-3a2cb2251ea3)
![Image](https://github.com/user-attachments/assets/03f90b70-b414-49a2-8794-51ab908f9d4a)
https://github.com/Ryang20718/cflag/pull/16

**Comment by jhecking** [comment]
It seems to finally work for us! But after many previous disappointments, I'll keep an eye on this to make sure it's not just a fluke but really a permanent solution.

![Image](https://github.com/user-attachments/assets/b82044ec-8417-4ff0-bb07-55b2770a9e4a)

**Comment by drazisil-codecov** [comment]
Hi @Ryang20718 , yes, the issue we found was the `--force` was not being passed though the entire flow to the place where it is checked for the check to pass or fail.

I'm surprised it worked for Jan and not you. That's encouraging, at least. 🤔 

@Swatinem This error was `AttributeError: 'NoneType' object has no attribute 'totals'"` and also missing "sessions"
SHA is `a606c4314528dc41381f78e2476f110d5b4ef5c9`

**Comment by drazisil-codecov** [comment]
@jhecking That's what I love to see! I completely understand your concerns. Please let me know if it stops again.

**Comment by Ryang20718** [comment]
@drazisil-codecov I created a new PR in my codebase and added this new empty upload feature and it does indeed work. I think rerunning my old PR just doesn't seem to work.

I think there's an edge case where this API doesn't work.

1. upload legimitate coverage, get results in PR for status checks
2. retrigger PR, but now modify codecov to upload empty report
3. status checks remain at `expecting` 

example https://github.com/Ryang20718/cflag/pull/24

**Comment by drazisil-codecov** [comment]
@Swatinem ^ <3

**Comment by Ryang20718** [comment]
Appreciate yall driving these issues down! 

please let me know if https://github.com/codecov/worker/pull/881 is supposed to fix this issue?

If not, please disregard this message below (still seems to be stuck on waiting to be reported) (I just re-triggered)
![Image](https://github.com/user-attachments/assets/75d8e414-5cc1-4a4c-affb-17cc9acc11f7)

**Comment by drazisil-codecov** [comment]
Hi @Ryang20718 It looks like checks were sent to `repos/Ryang20718/cflag/check-runs/33020188747` (API endpoint). I'm not having success locating where that maps to. GitHub did tell up `200`, though. 🤔 

edit:`af185d140ab573953fcaa0bcdd13a7a02b505e0d` is the correct head SHA, right?

**Comment by Ryang20718** [comment]
@drazisil-codecov `af185d140ab573953fcaa0bcdd13a7a02b505e0d` is the correct sha. this is the run https://github.com/Ryang20718/cflag/actions/runs/11749458250/job/33020182406?pr=24

PR link https://github.com/Ryang20718/cflag/pull/24

**Comment by drazisil-codecov** [comment]
@Ryang20718 Yes, I think the error that `af185d140ab573953fcaa0bcdd13a7a02b505e0d` had is fixed by https://github.com/codecov/worker/pull/894

Can you test again? (cc: @Swatinem

**Comment by Ryang20718** [comment]
@drazisil-codecov re-triggered with the same commit ![Image](https://github.com/user-attachments/assets/0b820d4d-9a32-4f6e-97ce-cad849362dd1)

Link to logs https://github.com/Ryang20718/cflag/actions/runs/11749458250/job/33058871228?pr=24

```
info - 2024-11-15 18:24:34,635 -- ci service found: github-actions
debug - 2024-[11](https://github.com/Ryang20718/cflag/actions/runs/11749458250/job/33058871228?pr=24#step:3:12)-15 18:24:34,638 -- versioning system found: <class 'codecov_cli.helpers.versioning_systems.GitVersioningSystem'>
debug - 2024-11-15 18:24:34,641 -- versioning system found: <class 'codecov_cli.helpers.versioning_systems.GitVersioningSystem'>
debug - 2024-11-15 18:24:34,643 -- Loading config from /home/runner/work/cflag/cflag/.codecov.yml
debug - 2024-11-15 18:24:34,653 -- Starting create commit process --- {"verbose": true, "auto_load_params_from": null, "codecov_yml_path": null, "enterprise_url": null, "version": "cli-0.8.0", "command": "create-commit", "parent_sha": null, "pull_request_number": "24", "branch": "testing_empty", "commit_sha": "af185d140ab573953fcaa0bcdd13a7a02b505e0d", "fail_on_error": false, "git_service": "github", "slug": "Ryang20718/cflag"}
info - 2024-11-15 18:24:35,186 -- Process Commit creating complete
debug - 2024-11-15 18:24:35,186 -- Commit creating result --- {"result": "RequestResult(error=None, warnings=[], status_code=202, text='{\"status\":\"queued\"}\\n')"}
info - 2024-11-15 18:24:35,927 -- ci service found: github-actions
debug - 2024-11-15 18:24:35,930 -- versioning system found: <class 'codecov_cli.helpers.versioning_systems.GitVersioningSystem'>
debug - 2024-11-15 18:24:35,933 -- versioning system found: <class 'codecov_cli.helpers.versioning_systems.GitVersioningSystem'>
debug - 2024-11-15 18:24:35,936 -- Loading config from /home/runner/work/cflag/cflag/.codecov.yml
debug - 2024-11-15 18:24:35,945 -- Starting create report process --- {"verbose": true, "auto_load_params_from": null, "codecov_yml_path": null, "enterprise_url": null, "version": "cli-0.8.0", "command": "create-report", "code": "default", "pull_request_number": "24", "commit_sha": "af185d140ab573953fcaa0bcdd13a7a02b505e0d", "fail_on_error": false, "git_service": "github", "slug": "Ryang20718/cflag"}
info - 2024-11-15 18:24:36,377 -- Process Report creating complete
debug - 2024-11-15 18:24:36,378 -- Report creating result --- {"result": "RequestResult(error=None, warnings=[], status_code=202, text='{\"status\":\"queued\"}\\n')"}
info - 2024-11-15 18:24:36,378 -- Finished creating report successfully --- {"response": "{\"status\":\"queued\"}\n"}
info - 2024-11-15 18:24:37,110 -- ci service found: github-actions
debug - 2024-11-15 18:24:37,113 -- versioning system found: <class 'codecov_cli.helpers.versioning_systems.GitVersioningSystem'>
debug - 2024-11-15 18:24:37,115 -- versioning system found: <class 'codecov_cli.helpers.versioning_systems.GitVersioningSystem'>
debug - 2024-11-15 18:24:37,118 -- Loading config from /home/runner/work/cflag/cflag/.codecov.yml
debug - 2024-11-15 18:24:37,[12](https://github.com/Ryang20718/cflag/actions/runs/11749458250/job/33058871228?pr=24#step:3:13)8 -- Starting empty upload process --- {"verbose": true, "auto_load_params_from": null, "codecov_yml_path": null, "enterprise_url": null, "version": "cli-0.8.0", "command": "empty-upload", "force": true, "commit_sha": "af185d140ab573953fcaa0bcdd[13](https://github.com/Ryang20718/cflag/actions/runs/11749458250/job/33058871228?pr=24#step:3:14)a7a02b505e0d", "fail_on_error": false, "git_service": "github", "slug": "Ryang20718/cflag"}
info - 2024-11-15 18:24:37,579 -- Process Empty Upload complete
debug - 2024-11-15 18:24:37,579 -- Empty Upload result --- {"result": "RequestResult(error=None, warnings=[], status_code=202, text='{\"status\":\"queued\"}\\n')"}
```

status checks are still stuck at waiting to be reported after rerunning the empty upload

**Comment by drazisil-codecov** [comment]
I've asked @suejung-sentry to take a look, as it's really late for Swatinem. It's getting kinda late, so we may have to pick this back up Monday, @Ryang20718 . I appreciate the patience and cooperation.

**Comment by drazisil-codecov** [comment]
Hi @Ryang20718 Looks like we are sending checks on the merge commit https://github.com/Ryang20718/cflag/runs/33058885826

https://github.com/Ryang20718/cflag/commit/af185d140ab573953fcaa0bcdd13a7a02b505e0d

We can see the checks here https://github.com/Ryang20718/cflag/pull/15 as well, when we expand the details.

Did you rerun the CI job?

Ideally we should have sent checks on https://github.com/Ryang20718/cflag/pull/15/commits/4abb84a3334a5bb88c4303b9fdf7cbf168cab983 as well, which it looks like we did. Is this finally behaving?

**Comment by Ryang20718** [comment]
@drazisil-codecov that's a different run which works for the happy path. The bug occurs when 

1. upload legimitate coverage, get results in PR for status checks
2. retrigger PR, but now modify codecov to upload empty report
3. status checks remain at `expecting` 

The run that I am mentioning is attempt # 3 https://github.com/Ryang20718/cflag/actions/runs/11749458250/attempts/3?pr=24

on [pull request # 24](https://github.com/Ryang20718/cflag/pull/24)

I re-triggered **another run** just now https://github.com/Ryang20718/cflag/actions/runs/11749458250/job/33151080963?pr=24 (1 minute ago.

**Comment by drazisil-codecov** [comment]
Hi @Ryang20718 

> retrigger PR, but now modify codecov to upload empty report

I think I'm confused here. How are you doing this? Any changes should be on a new SHA, are they not in this case?

**Comment by Ryang20718** [comment]
@drazisil-codecov 

github actions allows the ability to rerun workflows ![Image](https://github.com/user-attachments/assets/356a4015-26f1-466e-8e62-775de15abcd8)

I also made a new commit https://github.com/Ryang20718/cflag/pull/24/commits/358125049a7bbb69b287c78cfeaf040baaf41289 to retrigger CI to upload empty coverage, but that also has the same error

**Comment by drazisil-codecov** [comment]
Hi @Ryang20718 

I apologize, what i was unclear on was if you were saying you edut the workflow when yoy re-can it. The checks for https://github.com/Ryang20718/cflag/actions/runs/11897300171/job/33151232017#step:3:15 posted on https://github.com/Ryang20718/cflag/commit/39cd41c56562ddab13ce8cbdfa2dfa663066dca7, which is the SHA they were uploaded under.

Ypu may need to use the `--sha` flag https://docs.codecov.com/docs/cli-options#do-upload to tell Codedov what the correct SHA is, since it appears our autodetection is not picking the expected one.

**Comment by Ryang20718** [comment]
@drazisil-codecov That was the issue 👍 thanks!

**Comment by drazisil-codecov** [comment]
It is my hope that any further issues should be resolved by these two PRs

* https://github.com/codecov/codecov-api/pull/962
* https://github.com/codecov/engineering-team/issues/2536

It's not clear if these two PRs have deployed yet or not, since we are in the Sentry LaunchWeek code freeze. I will revisit this Monday and see if any further issues have been reported before I mark this done.

Please let me know if anyone is still seeing this (cc: @thomasrockhu-codecov

**Comment by drazisil-codecov** [comment]
@ all

There are quite a few customers on this thread. Before I close it, is anyone still seeing issues related to this?
